specification/authorization/resource-manager/Microsoft.Authorization/preview/2021-01-01-preview/authorization-RoleAssignmentApprovalCalls.json

```diff
@@ -0,0 +1,861 @@
+          "GetAccessReviews": {
+            "$ref": "./examples/ScopeGetRoleAssignmentApprovalSteps.json"
+          }
+        },
+        "description": "Get role assignment approval.",
+        "parameters": [
+          {
+            "name": "approvalId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The id of the role assignment approval."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "scope",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The scope of the resource.",
+            "x-ms-skip-url-encoding": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Describe the result of a successful operation.",
+            "schema": {
+              "$ref": "#/definitions/RoleAssignmentApprovalStepListResult"
+            }
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/ErrorDefinition"
+            }
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Authorization/roleAssignmentApprovals/{approvalId}/steps/{stepId}": {
+      "get": {
+        "tags": [
+          "ScopeGetRoleAssignmentApprovalStepById"
+        ],
+        "operationId": "ScopeRoleAssignmentApprovalStep_GetById",
+        "x-ms-examples": {
+          "GetAccessReviews": {
+            "$ref": "./examples/ScopeGetRoleAssignmentApprovalStepById.json"
+          }
+        },
+        "description": "Get role assignment approval.",
+        "parameters": [
+          {
+            "name": "approvalId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The id of the role assignment approval."
+          },
+          {
+            "name": "stepId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The id of the role assignment approval step."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "scope",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The scope of the resource.",
+            "x-ms-skip-url-encoding": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Describe the result of a successful operation.",
+            "schema": {
+              "$ref": "#/definitions/RoleAssignmentApprovalStep"
+            }
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/ErrorDefinition"
+            }
+          }
+        }
+      },
+      "patch": {
```

Comment:
>patch [](start = 7, length = 5)

PUT and PATCH verbs are both used for modifying an existing resource in ARM. (PUT is also used for creation, but that's not relevant in your case.) It's strange to support PATCH but not PUT.

You should add a PUT request for this as well. The difference is that PATCH allows a partial request body (usually just the property being modified) and PUT requires a full request body).

Reply:
Thanks for bringing this up! Appreciate it! 

I've added support for PUT as well.